File: app_banhang2/lib/services/models/model_order.dart
class ModelOrder {
  final String idUser;
  final List<Map<String, dynamic>> items;
  final double totalPrice;
  final double shippingFee;
  final double discount;
  final double finalTotal;
  final String paymentMethod;
  final String shippingAddress;
  final String? voucherId;

  ModelOrder({
    required this.idUser,
    required this.items,
    required this.totalPrice,
    required this.shippingFee,
    required this.discount,
    required this.finalTotal,
    required this.paymentMethod,
    required this.shippingAddress,
    this.voucherId,
  });

  Map<String, dynamic> toJson() => {
    'id_user': idUser,
    'items': items.map((item) => {
      'id_product': item['product']['_id'],
      'purchaseQuantity': item['purchaseQuantity'],
      'price': item['product']['price'],
    }).toList(),
    'totalPrice': totalPrice,
    'shippingFee': shippingFee,
    'discount': discount,
    'finalTotal': finalTotal,
    'paymentMethod': paymentMethod,
    'shippingAddress': shippingAddress,
    if (voucherId != null) 'voucherId': voucherId,
  };
}

File: APIBanHangReactNative/models/model_order.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const OrderSchema = new Schema({
    id_user: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    items: [{
        id_product: {
            type: Schema.Types.ObjectId,
            ref: 'products',
            required: true
        },
        purchaseQuantity: {
            type: Number,
            required: true
        },
        price: {
            type: Number,
            required: true
        }
    }],
    totalPrice: {
        type: Number,
        required: true
    },
    shippingFee: {
        type: Number,
        required: true
    },
    discount: {
        type: Number,
        default: 0
    },
    finalTotal: {
        type: Number,
        required: true
    },
    paymentMethod: {
        type: String,
        enum: ['cod', 'wallet'],
        required: true
    },
    shippingAddress: {
        type: String,
        required: true
    },
    status: {
        type: String,
        enum: [
            'pending',           // Chờ xác nhận
            'confirmed',         // Đã xác nhận
            'shipping',          // Đang giao hàng
            'delivered',         // Đã giao hàng
            'cancelled',         // Đã hủy
            'return_requested',  // Yêu cầu trả hàng
            'return_approved',   // Đã chấp nhận trả hàng
            'return_rejected',   // Từ chối trả hàng
            'returned',           // Đã hoàn trả
            'reviewed'  // Thêm trạng thái reviewed
        ],
        default: 'pending'
    },
    voucherId: {
        type: Schema.Types.ObjectId,
        ref: 'Voucher',
        required: false
    },
    returnRequest: {
        reason: String,
        images: [{
            type: String
        }],
        requestDate: Date,
        status: {
            type: String,
            enum: ['pending', 'approved', 'rejected'],
            default: 'pending'
        }
    }
}, { timestamps: true }); // Thêm timestamps


module.exports = mongoose.model('Order', OrderSchema);